lib/compiless.js

```diff
@@ -6,34 +6,60 @@ var Path = require('path'),
-            var ifNoneMatch = req.headers['if-none-match'];
-            if (ifNoneMatch) {
-                var validIfNoneMatchTokens = ifNoneMatch.split(" ").filter(function (etag) {
-                    return /-compiless\"$/.test(etag);
-                });
-                if (validIfNoneMatchTokens.length > 0) {
-                    req.headers['if-none-match'] = validIfNoneMatchTokens.join(" ");
-                } else {
-                    delete req.headers['if-none-match'];
-                }
+
+    return interceptor(function (req, res) {
+        // Prevent If-None-Match revalidation with the downstream middleware with ETags that aren't suffixed with "-compiless":
+        var ifNoneMatch = req.headers['if-none-match'];
+
+        if (ifNoneMatch) {
+            var validIfNoneMatchTokens = ifNoneMatch.split(" ").filter(function (etag) {
+                return /-compiless\"$/.test(etag);
+            });
+
+            if (validIfNoneMatchTokens.length > 0) {
+                req.headers['if-none-match'] = validIfNoneMatchTokens.join(" ");
+            } else {
+                delete req.headers['if-none-match'];
             }
-            delete req.headers['if-modified-since']; // Prevent false positive conditional GETs after enabling compiless
-            res.hijack(function (err, res) {
+        }
+        delete req.headers['if-modified-since']; // Prevent false positive conditional GETs after enabling compiless
+
+        return {
+            isInterceptable: function () {
+                return /\.less(?:\?.*)?$/.test(req.url);
+            },
+            intercept: function (body, send) {
+                // Prevent If-None-Match revalidation with the downstream middleware with ETags that aren't suffixed with "-compiless":
+                var ifNoneMatch = req.headers['if-none-match'];
+
+                if (ifNoneMatch) {
+                    var validIfNoneMatchTokens = ifNoneMatch.split(" ").filter(function (etag) {
+                        return /-compiless\"$/.test(etag);
+                    });
+
+                    if (validIfNoneMatchTokens.length > 0) {
+                        req.headers['if-none-match'] = validIfNoneMatchTokens.join(" ");
+                    } else {
+                        delete req.headers['if-none-match'];
+                    }
+                }
+
+                delete req.headers['if-modified-since']; // Prevent false positive conditional GETs after enabling compiless
+
                 var contentType = res.getHeader('Content-Type'),
                     matchContentType = contentType && contentType.match(/^text\/less(?:;\s*charset=([a-z0-9\-]+))?$/i);
+
                 // The mime module doesn't support less yet, so we fall back:
-                if (matchContentType || (/\.less(?:\?.*)?$/.test(req.url) && contentType === 'application/octet-stream')) {
+                if (matchContentType || contentType === 'application/octet-stream') {
```

Comment:
Yes indeed! that is checking that the file is actually a .less file

Now this validation is being added to the function isInterceptable from express-interceptor. So if it's not a .less file the response just gets ignore.